pkg/faces/constants.go

```diff
@@ -17,42 +17,102 @@
 
 package faces
 
-var Smileys = map[string]string{
-	"Smiling":     "&#x1F603;",
-	"Sleeping":    "&#x1F634;",
-	"Cursing":     "&#x1F92C;",
-	"Kaboom":      "&#x1F92F;",
-	"HeartEyes":   "&#x1F60D;",
-	"Neutral":     "&#x1F610;",
-	"RollingEyes": "&#x1F644;",
-	"Screaming":   "&#x1F631;",
+type SmileyMap struct {
+	smileys map[string]string
 }
 
-// colorblind-friendly colors from the Tol palette
-var Colors = map[string]string{
-	"grey":    "grey",
-	"purple":  "rgb(48 34 130)",
-	"green":   "rgb(55 117 59)",
-	"cyan":    "rgb(55 117 59)",   // too similar to pink and grey, avoid
-	"blue":    "rgb(151 202 234)", // light blue
-	"yellow":  "rgb(218 204 130)",
-	"pink":    "rgb(191 108 120)",
-	"hotpink": "rgb(158 75 149)",
-	"red":     "rgb(125 42 83)", // dark magenta
+var Smileys = SmileyMap{
+	smileys: map[string]string{
+		"Grinning":    "&#x1F603;",
+		"Sleeping":    "&#x1F634;",
+		"Cursing":     "&#x1F92C;",
+		"Kaboom":      "&#x1F92F;",
+		"HeartEyes":   "&#x1F60D;",
+		"Neutral":     "&#x1F610;",
+		"RollingEyes": "&#x1F644;",
+		"Screaming":   "&#x1F631;",
+		"Vomiting":    "&#x1F92E;",
+	},
+}
+
+// Lookup a smiley by name. If found, return the HTML entity for the smiley
+// and true; if not found, return an empty string and false.
+func (sm *SmileyMap) Lookup(name string) (string, bool) {
+	if smiley, ok := sm.smileys[name]; ok {
+		return smiley, true
+	}
+
+	return "", false
+}
+
+type Palette struct {
+	colors map[string]string
+}
+
+// These colors are from https://davidmathlogic.com/colorblind/, using the
+// "Tol" palette to be colorblindness-friendly. These are RGB values from
+// top to bottom in the palette as shown on that page.
+
+var Colors = Palette{
+	colors: map[string]string{
+		// Include grey/black/white because they're sometimes convenient.
+		"grey":      "grey",
+		"black":     "black",
+		"white":     "white",
+
+		// These are colorblindness-friendly values from the "Tol" palette, as
+		// noted above.
+		"violet":    "#332288",
+		"green":     "#117733",
+		"bluegreen": "#44AA99",	// best avoided, see below
+		"blue":      "#88CCEE",
+		"yellow":    "#DDCC77",
+		"coral":     "#CC6677",	// best avoided, see below
+		"fuschia":   "#AA4499",
+		"red":       "#882255", // be careful, see below
+
+		// Bluegreen and coral should be avoided, full stop, since they are very
+		// similar to each other for folks with protanopia or deuteranopia.
+		//
+		// The red/green pair and the green/blue pair can be similar enough for
+		// folks with some forms of colorblindness that it might be a bad idea to
+		// rely on just the color for semantics: e.g. if red means "bad" and green
+		// means "good", it might not be immediately apparently what's up if the
+		// display shows all red or all green. Red/cyan and green/violet are much
```

Comment:
Unclear what color `cyan` is referring to - is it `blue` or `bluegreen`? Should fix these names for consistency.